### PR TITLE
Make tests faster in Miri

### PR DESCRIPTION
**Pull request by RalfJung**
Reduce some test iteration counts in Miri.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
r=me, unless you want to wait for the investigation to conclude

File: library/alloc/tests/slice.rs [hunk 1]
```diff
@@ -1722,7 +1722,7 @@ fn panic_safe() {
     let mut rng = thread_rng();
 
     // Miri is too slow
-    let lens = if cfg!(miri) { (1..10).chain(20..21) } else { (1..20).chain(70..MAX_LEN) };
+    let lens = if cfg!(miri) { (1..10).chain(0..0) } else { (1..20).chain(70..MAX_LEN) };
```
Comment (RalfJung) [inline library/alloc/tests/slice.rs:1725]: I added the "20" case without measuring, which [clearly was a mistake](https://github.com/RalfJung/miri-test-libstd/issues/2). So let's remove it again.

(We have to `chain` to make this type-check.)

**Comment by RalfJung** [comment]
Investigation concluded, but it made me realize I should also check the libcore test suite for overlong tests, so I am doing that now.

**Comment by RalfJung** [comment]
@Mark-Simulacrum I also adjusted some libcore tests.

File: library/alloc/tests/slice.rs (hunk 1 above)
Comment (NieDzejkob) [reply library/alloc/tests/slice.rs:1725]: > (We have to `chain` to make this type-check.)

Perhaps this rationale is best left in a comment?

Comment (RalfJung) [reply library/alloc/tests/slice.rs:1725]: Good point, done.

**Pull request merged**